beacon_node/network/src/sync/sampling.rs

```diff
@@ -433,7 +433,6 @@ mod request {
         column_index: ColumnIndex,
         status: Status,
         // TODO(das): Should downscore peers that claim to not have the sample?
-        #[allow(dead_code)]
         peers_dont_have: HashSet<PeerId>,
```

Comment:
Not having is not a permanent state, so we should move to a timed hash set. But can be done in another PR

Reply:
yeah agree, this needs a bit more thought to handle edge cases like this (peer didn't have it but have it now) and make this reliable under real network conditions.

Reply:
Created issue https://github.com/sigp/lighthouse/issues/6110